src/modules/fancyzones/lib/FancyZonesData.h

```diff
@@ -0,0 +1,137 @@
+        {
+            std::scoped_lock lock{ dataLock };
+            return customZoneSetsMap;
+        }
+
+        inline const std::unordered_map<std::wstring, std::vector<FancyZonesDataTypes::AppZoneHistoryData>>& GetAppZoneHistoryMap() const
+        {
+            std::scoped_lock lock{ dataLock };
+            return appZoneHistoryMap;
+        }
+
+        void AddDevice(const std::wstring& deviceId);
+        void CloneDeviceInfo(const std::wstring& source, const std::wstring& destination);
+        void UpdatePrimaryDesktopData(const std::wstring& desktopId);
+        void RemoveDeletedDesktops(const std::vector<std::wstring>& activeDesktops);
+
+        bool IsAnotherWindowOfApplicationInstanceZoned(HWND window, const std::wstring_view& deviceId) const;
+        void UpdateProcessIdToHandleMap(HWND window, const std::wstring_view& deviceId);
+        std::vector<int> GetAppLastZoneIndexSet(HWND window, const std::wstring_view& deviceId, const std::wstring_view& zoneSetId) const;
+        bool RemoveAppLastZone(HWND window, const std::wstring_view& deviceId, const std::wstring_view& zoneSetId);
+        bool SetAppLastZones(HWND window, const std::wstring& deviceId, const std::wstring& zoneSetId, const std::vector<int>& zoneIndexSet);
+
+        void SetActiveZoneSet(const std::wstring& deviceId, const FancyZonesDataTypes::ZoneSetData& zoneSet);
+
+        bool SerializeDeviceInfoToTmpFile(const std::wstring& uniqueId) const;
+        void ParseDataFromTmpFiles();
+
+        json::JsonObject GetPersistFancyZonesJSON();
+
+        void LoadFancyZonesData();
+        void SaveFancyZonesData() const;
+
+    private:
+#if defined(UNIT_TESTS)
+        friend class FancyZonesUnitTests::FancyZonesDataUnitTests;
+        friend class FancyZonesUnitTests::FancyZonesIFancyZonesCallbackUnitTests;
+        friend class FancyZonesUnitTests::ZoneWindowUnitTests;
+        friend class FancyZonesUnitTests::ZoneSetCalculateZonesUnitTests;
+
+        inline void SetDeviceInfo(const std::wstring& deviceId, FancyZonesDataTypes::DeviceInfoData data)
+        {
+            deviceInfoMap[deviceId] = data;
+        }
+
+        inline bool ParseDeviceInfos(const json::JsonObject& fancyZonesDataJSON)
+        {
+            deviceInfoMap = JSONHelpers::ParseDeviceInfos(fancyZonesDataJSON);
+            return !deviceInfoMap.empty();
+        }
+
+        inline void clear_data()
+        {
+            appZoneHistoryMap.clear();
+            deviceInfoMap.clear();
+            customZoneSetsMap.clear();
+        }
+
+        inline void SetSettingsModulePath(std::wstring_view moduleName)
+        {
+            std::wstring result = PTSettingsHelper::get_module_save_folder_location(moduleName);
+            zonesSettingsFileName = result + L"\\" + std::wstring(L"zones-settings.json");
+            appZoneHistoryFileName = result + L"\\" + std::wstring(L"app-zone-history.json");
```

Comment:
As in one of the previous comments: replace all string literals with const strings and explicitly mark them as non-localizable.